Source/WebCore/storage/StorageNamespaceProvider.h

```diff
@@ -46,15 +46,22 @@ class StorageNamespaceProvider : public RefCounted<StorageNamespaceProvider> {
     WEBCORE_EXPORT StorageNamespaceProvider();
     WEBCORE_EXPORT virtual ~StorageNamespaceProvider();
 
-    virtual Ref<StorageNamespace> createSessionStorageNamespace(Page&, unsigned quota) = 0;
-
     Ref<StorageArea> localStorageArea(Document&);
+    Ref<StorageArea> sessionStorageArea(Document&);
+
+    enum class ShouldCreateIfNotExists : bool { No, Yes };
+    virtual RefPtr<StorageNamespace> sessionStorageNamespace(SecurityOrigin&, Page&, ShouldCreateIfNotExists = ShouldCreateIfNotExists::Yes) = 0;
 
     WEBCORE_EXPORT void setSessionIDForTesting(PAL::SessionID);
 
+    void setSessionStorageQuota(unsigned quota) { m_sessionStorageQuota = quota; }
+    virtual void setSessionStorageNamespace(SecurityOrigin&, Page&, RefPtr<StorageNamespace>&&) = 0;
```

Comment:
Can we pass the origin as a `const SecurityOrigin&` ?

Reply:
Done.